docs/design/ttl_scans.md

```diff
@@ -0,0 +1,79 @@
+# TTL scans
+
+## Summary
+
+Add an option to automatically delete old starboard scans, in this first version focus on vulnerability scans but in the long run we could add similar functionality to other reports as well.
+
+## Motivation
+
+In [537](https://github.com/aquasecurity/starboard/issues/537) we talk about a need to run nightly vulnerability scans of CVE:s.
+This way we can make sure to get new CVE reports for long time running pods as well.
+
+## Proposal
+
+Add a environment variable to the operator, for example `OPERATOR_TTL_VULNERABILITY_SCANNER=24h`, this way we can add other ttl values for other reports as well.
+Adding this environment variable would add a annotation to the generated VulnerabilityReport that we can look for.
+
+Using similar functionality as defined in [ttlAfterFinished](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/) for cronjob seems to be the most versatile option.
+
+Just like the [ttlAfterFinished controller](https://github.com/kubernetes/kubernetes/blob/2a821d787b7f21fbb399b0a0bf6d3fa0c79f4293/pkg/controller/ttlafterfinished/ttlafterfinished_controller.go) we should add it as a separate controller.
+This way it will be easy to extend to other reports as well.
+
+If the operator restarts the queue will disappear since we don't hold any state so we would have to scan all the available reports and calculate
+the ttl comparing it to `report.updateTimestamp`.
+
+We could calculate the ttl without having creating a new annotation to the reports but the verbosity of showing the users how long each report
+got a ttl outweighs the "issue" of generating a new annotation.
+
+### Example
+
+Below you can see a shortened version of the yaml. Notice the `metadata.annotations.ttlVulnerabilityReport` which is the new.
+The operator would automatically apply the `ttlVulnerabilityReport` annotation to all reports that it generates assuming that the environment variable is set.
```

Comment:
According to [Syntax and character set](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set) an annotation has prefix and name to avoid collisions.

Maybe we could use `starboard.aquasecurity.github.io/report-ttl` for any kind of security report managed by Starboard. WDYT?

Reply:
I know about this but for some reason I didn't think about it when writing the example... good catch and I have updated the PR accordingly.